### PR TITLE
 TCIOS-808488 Developer should be able to set custom requestNonce for verification request

### PR DESCRIPTION
**Pull request by sreedeepkesav**

### CODE REVIEW

File: TrueSDK/External/TCTrueSDK.h [hunk 1]
```diff
@@ -76,6 +76,16 @@
 - (void)setupWithAppKey:(nonnull NSString *)appKey
                 appLink:(nonnull NSString *)appLink;
 
+/*!
+* @brief Setup by providing the Partner Key and the App Link. SDK needs to be set up before any request.
+* @param appKey Your App Key provided by Truecaller
+* @param appLink Your App Link url string provided by Truecaller
+* @param requestNonce Your App generrated custom nonce
```
Comment (elnaqah) [inline TrueSDK/External/TCTrueSDK.h]: typo generated

File: TrueSDK/External/TCTrueSDK.h [hunk 2]
```diff
@@ -76,6 +76,16 @@
...
+*/
+- (void)setupWithAppKey:(nonnull NSString *)appKey
+                appLink:(nonnull NSString *)appLink
+           requestNonce:(nonnull NSString *)requestNonce;
```
Comment (elnaqah) [inline TrueSDK/External/TCTrueSDK.h]: as this is optional should be nullable

Comment (sreedeepkesav) [reply TrueSDK/External/TCTrueSDK.h]: Done

File: TrueSDK/External/TCTrueSDK.m [hunk 3]
```diff
@@ -80,7 +90,7 @@ - (void)requestTrueProfile
         return;
     }
     
-    NSString *requestNonce = [NSUUID UUID].UUIDString;
+    NSString *requestNonce = self.requestNonce != nil ? self.requestNonce : [NSUUID UUID].UUIDString;
```
[the remark this answers is not in the document]
Comment (sreedeepkesav) [reply TrueSDK/External/TCTrueSDK.m]: Done

**Pull request merged**